Pokedex_ios/View/Controller/PokemonListViewController.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  ViewController.swift
+//  Pokedex_ios
+//
+//  Created by admin on 6/29/22.
+//
+
+import UIKit
+
+class PokemonListViewController: UIViewController {
+    
+    @IBOutlet weak var tableView: UITableView!
+    
+    lazy var viewModel = {
+        PokemonListViewModel()
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tableView.delegate = self
+        tableView.dataSource = self
+        setBackground()
+        initViewModel()
+        
+        let uiNib = UINib(nibName: "PokemonTableViewCell", bundle: nil)
+        tableView.register(uiNib, forCellReuseIdentifier: "PokeCell")
+    }
+    
+    private func setBackground() {
+        let margins = view.layoutMarginsGuide
+        
+        let imageView = UIImageView()
+        imageView.image = UIImage(named: "PokeballSVG")
+        imageView.alpha = 0.03
+        view.addSubview(imageView)
+        
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.topAnchor.constraint(equalTo: margins.topAnchor, constant: -300).isActive = true
+        imageView.widthAnchor.constraint(equalTo: view.widthAnchor, constant: 1).isActive = true
+        imageView.heightAnchor.constraint(equalTo: imageView.widthAnchor, constant: 1).isActive = true
+        
+        self.view.sendSubviewToBack(imageView)
+    }
+    
+    func initViewModel() {
+        viewModel.getPokemons()
+        print(viewModel.pokemons)
+        
+        viewModel.reloadTableView = { [weak self] in
```

Comment:
ReloadData should be called on *getPokemons()* success completion. View model should handle all the businnes logic only

Reply:
Ok maybe changing the function's name will be better